src/main/java/net/sf/jabref/logic/specialfields/SpecialField.java

```diff
@@ -51,8 +50,6 @@ protected void setValues(List<SpecialFieldValue> values) {
 
```

Comment:
What @tobiasdiez means with the enum solution is to turn this class into an enum and merge all of the different special field classes (`Priority`, `Printed`, etc.) into this one. This seems reasonable, since all special fields are singletons anyway (you always access them via `getInstance`). You loose the inheritance relation, but apart from that enums can also have arbitrary methods, constructors, just as normal classes. Here is some explanation of how this works: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

The problem that this raises, however, is that all special field classes would move into `SpecialField` and, thereby, also `SpecialFieldValue` would be needed there. To enable this, we have to move the gui stuff out of `SpecialFieldValue` and that is at least the `Icon` in there. In the end, this class really needs to be broken up into the parts that are gui independent and the parts that are gui dependent. I understand that this is not easy.